.github/actions/deb-delivery/action.yml

```diff
@@ -96,9 +96,17 @@ runs:
           # build a search pattern using ARTIFACT_BASENAME+OS_DISTRIB+ARCH
           VERSION=${{ inputs.major_version }}
           BASENAME=$(echo "$file" | cut -d '_' -f 1)
-          OS_DISTRIB=$(echo "$file" | sed -n 's/.*+\(deb[0-9]\+\).*/\1/p')
+          OS_DISTRIB=""
           ARCH=$(echo $file | cut -d '_' -f3 | cut -d '.' -f1)
 
+          if [[ "${{ inputs.distrib }}" == "jammy" ]] || [[ "${{ inputs.distrib }}" == "noble" ]]; then
+            OS_DISTRIB=$(echo "$file" | sed -nE 's/.*(ubuntu\.[0-9]{2}\.[0-9]{2}).*/\1/p')
+          elif [[ "${{ inputs.distrib }}" == "bullseye" ]] || [[ "${{ inputs.distrib }}" == "bookworm" ]]; then
+            OS_DISTRIB=$(echo "$file" | sed -nE 's/.*(deb[0-9]+).*/\1/p')
```

Comment:
wouldn t this remain closer to the original +deb12 pattern ?
`.*(\+deb[0-9]+).*`